event/listener.go

```diff
@@ -23,27 +40,19 @@ type Listener interface {
 	Stop() error
 }
 
-func NewListener(processor Processor, manager Manager, subID string, buffer int, filters ...Filter) Listener {
+func NewListener(manager Manager, handlers ...*Handler) Listener {
 	return &listener{
-		processor: processor,
 		manager:   manager,
-		subID:     subID,
-		buffer:    buffer,
-		filters:   filters,
 		lock:      &sync.Mutex{},
 		waitGroup: &sync.WaitGroup{},
+		handlers:  handlers,
 	}
 }
 
 type listener struct {
```

Comment:
I don't understand what is the goal of the listener now that it delegates processing logic to a handler.
It just manages multiple handlers that aren't necessarily logically combined, meaning we can have a single listener for the whole app. 

Why do we need handlers separated from the listener ?

Reply:
It's all about a start/stop service that needs to be injected into a root app builder. This way we have one global listener. Handler is simply a DTO to bring toghether processor, filters and buffering configuration.